plugin/distribution/pull.go

```diff
@@ -153,7 +153,8 @@ func Pull(ref reference.Named, rs registry.Service, metaheader http.Header, auth
 		logrus.Debugf("pull.go: error in json.Unmarshal(): %v", err)
 		return nil, err
 	}
-	if m.Config.MediaType != schema2.MediaTypePluginConfig {
+	if m.Config.MediaType != schema2.MediaTypePluginConfig &&
+		m.Config.MediaType != "application/vnd.docker.plugin.image.v0+json" { //TODO: remove this v0 before 1.13 GA
```

Comment:
A similar check for `image.v0` needs to be added in https://github.com/vieux/docker/blob/exit_exp_plugin/distribution/pull_v2.go#L361, to avoid using `docker pull` to pull plugin images.